extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/health/ReactiveDB2DataSourceHealthCheck.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.reactive.db2.client.runtime.health;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import io.quarkus.arc.Arc;
+import io.vertx.db2client.DB2Pool;
+
+@Readiness
+@ApplicationScoped
+public class ReactiveDB2DataSourceHealthCheck implements HealthCheck {
+    private DB2Pool db2Pool;
+
+    @PostConstruct
+    protected void init() {
+        db2Pool = Arc.container().instance(DB2Pool.class).get();
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Reactive DB2 connection health check").up();
+
+        try {
+            CompletableFuture<Void> databaseConnectionAttempt = new CompletableFuture<>();
```

Comment:
Using the Mutiny version would simplify this to:

`pool.query(...).await().atMost(Duration.ofSeconds(10))`